codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -48,8 +49,16 @@
  * ...
  * p.addLast("handler", new HttpRequestHandler());
  * </pre>
- * Be aware that you need to have the {@link HttpResponseEncoder} or {@link HttpRequestEncoder}
- * before the {@link HttpObjectAggregator} in the {@link ChannelPipeline}.
+ * Be aware that you need to have the {@link HttpResponseDecoder} or {@link HttpRequestDecoder}
+ * before the {@link HttpObjectAggregator} in the {@link ChannelPipeline} for it to receive
+ * {@link HttpObject}s to aggregate, and you need a {@link HttpResponseEncoder} before the
+ * {@link HttpObjectAggregator} if it will be receiving {@link HttpRequest} messages as the
+ * {@link HttpObjectAggregator} may write a 100 Continue response.
```

Comment:
> This makes it sounds like HttpResponseEncoder must be before HttpObjectAggregator bcz HttpObjectAggregator is responsible for sending 100-continue responses.

Isn't that true though?  `HttpResponseEncoder` must be before `HttpObjectAggregator` when a `HttpRequest` is sent to HttpObjectAggregator (aka, when a HttpRequestDecoder is before it) as the HttpObjectAggregator can respond with a 100, 413, or 417 depending on the circumstances. 

Maybe the better thing to do is to avoid repeating the advice of what must be before what ([already mentioned above](https://github.com/netty/netty/pull/5698/files/727ee18f895d5702c0ab8623cd84b9d667b6794d#diff-d88515eae60a5883db7d8d08726ac350R41)), and instead focus on what else HttpObjectAggregator can do for you (sending 100, 413, or 417 responses).  That would let someone piece together why the response encoder is needed without needing to spell it out in a repetitive way.

What I was trying to explain here was that having the HttpResponseEncoder is conditional on there being a HttpRequestDecoder.  If you need one then you need the other, but if you are using a HttpResponseDecoder then you don't need any Encoder (at least not for HttpObjectAggregator).  The paragraph after this one could help someone piece that together, so maybe that level of detail isn't needed after all.